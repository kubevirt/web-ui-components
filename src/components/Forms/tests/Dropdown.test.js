import createTestContext from '../../../cosmos/enzyme';
import fixture from '../fixtures/Dropdown.fixture';

const { mount, getWrapper } = createTestContext({ fixture });

beforeEach(mount);

describe('<DropdownControl />', () => {
  it('renders correctly', () => {
    expect(getWrapper()).toMatchSnapshot();
  });
});
