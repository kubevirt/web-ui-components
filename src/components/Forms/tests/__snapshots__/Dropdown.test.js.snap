// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownControl /> renders correctly 1`] = `
<Dropdown
  choices={
    Array [
      Object {
        "name": "choice1",
      },
      Object {
        "name": "choice2",
      },
    ]
  }
  id="1"
  onBlur={null}
  onChange={[MockFunction]}
  value="This is dropdown button"
>
  <ButtonGroup
    block={false}
    bsClass="btn-group"
    justified={true}
    key="1"
    vertical={false}
  >
    <div
      className="btn-group btn-group-justified"
    >
      <DropdownButton
        bsStyle="default"
        className="form-dropdown"
        id="1"
        onBlur={null}
        onSelect={[Function]}
        title="This is dropdown button"
      >
        <DropdownButton
          bsStyle="default"
          className="form-dropdown"
          id="1"
          onBlur={null}
          onSelect={[Function]}
          title="This is dropdown button"
        >
          <Uncontrolled(Dropdown)
            bsStyle="default"
            id="1"
            onSelect={[Function]}
          >
            <Dropdown
              bsClass="dropdown"
              bsStyle="default"
              componentClass={[Function]}
              id="1"
              onSelect={[Function]}
              onToggle={[Function]}
            >
              <ButtonGroup
                block={false}
                bsClass="btn-group"
                bsStyle="default"
                className="dropdown"
                justified={false}
                vertical={false}
              >
                <div
                  className="dropdown btn-group btn-group-default"
                >
                  <DropdownToggle
                    bsClass="dropdown-toggle"
                    bsRole="toggle"
                    bsStyle="default"
                    className="form-dropdown"
                    id="1"
                    key=".0"
                    onBlur={null}
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    open={false}
                    useAnchor={false}
                  >
                    <Button
                      active={false}
                      aria-expanded={false}
                      aria-haspopup={true}
                      block={false}
                      bsClass="btn"
                      bsStyle="default"
                      className="form-dropdown dropdown-toggle"
                      disabled={false}
                      id="1"
                      onBlur={null}
                      onClick={[Function]}
                      onKeyDown={[Function]}
                      role="button"
                    >
                      <button
                        aria-expanded={false}
                        aria-haspopup={true}
                        className="form-dropdown dropdown-toggle btn btn-default"
                        disabled={false}
                        id="1"
                        onBlur={null}
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        type="button"
                      >
                        This is dropdown button
                         
                        <span
                          className="caret"
                        />
                      </button>
                    </Button>
                  </DropdownToggle>
                  <DropdownMenu
                    bsClass="dropdown-menu"
                    bsRole="menu"
                    key=".1"
                    labelledBy="1"
                    onClose={[Function]}
                    onSelect={[Function]}
                    pullRight={false}
                  >
                    <RootCloseWrapper
                      disabled={true}
                      event="click"
                      onRootClose={[Function]}
                    >
                      <ul
                        aria-labelledby="1"
                        className="dropdown-menu"
                        role="menu"
                      >
                        <MenuItem
                          bsClass="dropdown"
                          disabled={false}
                          divider={false}
                          eventKey="choice1"
                          header={false}
                          key=".$choice1"
                          onKeyDown={[Function]}
                          onSelect={[Function]}
                        >
                          <li
                            className=""
                            role="presentation"
                          >
                            <SafeAnchor
                              componentClass="a"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                            >
                              <a
                                href="#"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                role="menuitem"
                                tabIndex="-1"
                              >
                                choice1
                              </a>
                            </SafeAnchor>
                          </li>
                        </MenuItem>
                        <MenuItem
                          bsClass="dropdown"
                          disabled={false}
                          divider={false}
                          eventKey="choice2"
                          header={false}
                          key=".$choice2"
                          onKeyDown={[Function]}
                          onSelect={[Function]}
                        >
                          <li
                            className=""
                            role="presentation"
                          >
                            <SafeAnchor
                              componentClass="a"
                              onClick={[Function]}
                              onKeyDown={[Function]}
                              role="menuitem"
                              tabIndex="-1"
                            >
                              <a
                                href="#"
                                onClick={[Function]}
                                onKeyDown={[Function]}
                                role="menuitem"
                                tabIndex="-1"
                              >
                                choice2
                              </a>
                            </SafeAnchor>
                          </li>
                        </MenuItem>
                      </ul>
                    </RootCloseWrapper>
                  </DropdownMenu>
                </div>
              </ButtonGroup>
            </Dropdown>
          </Uncontrolled(Dropdown)>
        </DropdownButton>
      </DropdownButton>
    </div>
  </ButtonGroup>
</Dropdown>
`;
